packages/default-reporter/src/reporterForClient/reportProgress.ts

```diff
@@ -156,16 +172,41 @@ function getProgessStatsPushStreamByRequirer (progress$: most.Stream<ProgressLog
   return progessStatsPushStreamByRequirer
 }
 
+function getPackageImportMethodPushStream (packageImportMethod$: most.Stream<PackageImportMethodLog>) {
+  const packageImportMethodPushStream = new PushStream<string>()
+
+  let packageImportMethod = 'hard linked'
+  packageImportMethod$
+    .forEach((log: PackageImportMethodLog) => {
+      switch (log.method) {
+        case 'copy':
+          packageImportMethod = 'copied'
+          break
+        case 'clone':
+          packageImportMethod = 'cloned'
+          break
+        default:
+          break
+      }
+      packageImportMethodPushStream.next(packageImportMethod)
+    })
+
+  return packageImportMethodPushStream
+}
+
 function createStatusMessage (
+  showImportingMethod: boolean,
   progress: ProgressStats,
+  importingMethod: string,
   importingDone: boolean
 ) {
   const msg = `Resolving: total ${hlValue(progress.resolved.toString())}, reused ${hlValue(progress.reused.toString())}, downloaded ${hlValue(progress.fetched.toString())}`
   if (importingDone) {
+    const importingMsg = `Packages were ${importingMethod} from the content-addressable store to the virtual store.\nContent-addressable store is at: ~/.pnpm-store/v3\nVirtual store is at: node_modules/.pnpm`
```

Comment:
we should probably log the real used store location and virtual store locations. Unfortunatly, they are not in the config.

They are only set in this file: https://github.com/pnpm/pnpm/blob/98ed2544d28582fae83f983eef796365dddc6a01/packages/get-context/src/index.ts#L85

maybe we could add some new log, something like contextLogger that would contain information like the storeDir, virtualStoreDir, currentLockfileExists.

Reply:
I like the idea of contextLogger